scripts/build.js

```diff
@@ -31,9 +35,20 @@ if (glob) {
   rimraf.sync(
     `./{packages,plugins}/${glob}/{lib,esm,tsconfig.tsbuildinfo,node_modules/@types/react}`,
   );
-  const packageName = glob.replace(/^superset-ui-/, '');
-  run(`nimbus babel --clean --workspaces="@superset-ui/${packageName}" ${BABEL_CONFIG}`);
-  run(`nimbus babel --clean --workspaces="@superset-ui/${packageName}" --esm ${BABEL_CONFIG}`);
+  let packageName = glob.replace(/^superset-ui-/, '');
+  if (!extraArgs.includes('--type-only')) {
+    run(`nimbus babel --clean --workspaces="@superset-ui/${packageName}" ${BABEL_CONFIG}`);
+    run(`nimbus babel --clean --workspaces="@superset-ui/${packageName}" --esm ${BABEL_CONFIG}`);
+  }
+  // only build for packages with ts files
+  packageName = `(${[
+    ...new Set(
+      fastGlob
+        .sync([`./node_modules/@superset-ui/${glob}/src/**/*.ts`])
+        .map(x => x.split('/')[3])
+        .filter(x => !metaPackages.has(x)),
+    ),
+  ].join('|')})`;
   run(`nimbus typescript --build --workspaces="@superset-ui/${packageName}"`);
```

Comment:
By using `--workspaces` instead of `--reference-workspaces`, nimbus is able to run `tsc` in parallel and significantly increases building speed (5min -> 1min).

Reply:
Had to do this initially because `nimbus typescript --reference-workspaces` (which generates different `tsconfig.json`) somehow couldn't find `jed` defined in `external.d.ts` no matter what.

Reply:
🥇